library/x509_crt.c

```diff
@@ -524,6 +524,11 @@ static int x509_get_basic_constraints( unsigned char **p,
         return( MBEDTLS_ERR_X509_INVALID_EXTENSIONS +
                 MBEDTLS_ERR_ASN1_LENGTH_MISMATCH );
 
+    // Do not accept max_pathlen equal to INT_MAX due to size constraints
```

Comment:
What are the size constraints? Isn't this to avoid undefined behaviour when we would do signed integer overflow on line 532 below?

Reply:
Yes it is, maybe calling this a size constraint isn't precise enough? :)